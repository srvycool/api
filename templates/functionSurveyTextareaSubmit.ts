export const request = `
  #set($submissionID = $util.autoId())

  {
    "version" : "2017-02-28",
    "operation" : "PutItem",
    "key" : {
      "node" : $util.dynamodb.toDynamoDBJson($ctx.stash.surveyID),
      "path" : $util.dynamodb.toDynamoDBJson("submission#$submissionID")
    },
    "attributeValues" : {
      "id": $util.dynamodb.toDynamoDBJson($submissionID),
      "text": $util.dynamodb.toDynamoDBJson($ctx.stash.text)
    }
  }
`;

export const response = `
  {}
`;
