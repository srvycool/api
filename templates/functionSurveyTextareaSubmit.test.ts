jest.mock("uuid", () => ({ v4: () => "gg-hh-ii" }));

import Parser, { Context } from "appsync-template-tester";
import { request, response } from "./functionSurveyTextareaSubmit";

describe("templates:functionSurveyTextareaSubmit", () => {
  describe("request", () => {
    it("should match snapshot", () => {
      const parser = new Parser(request);
      const context = {
        stash: {
          surveyID: "aa-bb-cc",
          text: "Awesome!",
        },
      } as Context;

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe("response", () => {
    it("should match snapshot", () => {
      const parser = new Parser(response);
      const context = {};

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});
