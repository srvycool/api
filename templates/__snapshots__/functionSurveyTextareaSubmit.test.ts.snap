// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates:functionSurveyTextareaSubmit request should match snapshot 1`] = `
Object {
  "attributeValues": Object {
    "id": Object {
      "S": "gg-hh-ii",
    },
    "text": Object {
      "S": "Awesome!",
    },
  },
  "key": Object {
    "node": Object {
      "S": "aa-bb-cc",
    },
    "path": Object {
      "S": "submission#gg-hh-ii",
    },
  },
  "operation": "PutItem",
  "version": "2017-02-28",
}
`;

exports[`templates:functionSurveyTextareaSubmit response should match snapshot 1`] = `Object {}`;
